File: replaceforcolumnname2.py
import pandas as pd

df = pd.DataFrame({
    'EmbarkedS': [1,2,3,4,5],
    'EmbarkedQ' : [1,2,3,4,5],
    'EmbarkedR' : [1,2,3,4,5]
})

print(df)
# df.columns = df.columns.str.replace('Embarked', ' ')
# print(df)
# print(df[' S'])


df.columns = df.columns.str.replace('Embarked', ' ').str.strip()
print(df)
print(df['S'])

File: replaceforcolumnname2.r
library(tidyverse)

df = data.frame(
    EmbarkedS = c(1,2,3,4,5),
    EmbarkedQ = c(1,2,3,4,5),
    EmbarkedR = c(1,2,3,4,5)
)
print(df)
colnames(df) <- gsub('Embarked', '', colnames(df))
print(df)